File: SDKPackages/da/android/SampleCode/DigitalAnalyticsKitchenSink/app/src/main/java/com/tl/digitalanalytics/tagging/TagProductView.java
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2016
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 ******************************************************************************/
package com.tl.digitalanalytics.tagging;

import android.app.Activity;
import android.util.Log;

import com.da.kitchensink.R;
import com.digitalanalytics.api.DigitalAnalytics;
import com.tl.digitalanalytics.model.Product;

/**
 * 
 * @author Sohil Shah (sohishah@us.ibm.com)
 * 
 */
public class TagProductView {
	private Activity activity;
	private Product[] products;

	public TagProductView(final Activity activity, Product[] products) {
		this.activity = activity;
		this.products = products;
	}

	public void executeTag() {
		try {

			Boolean success = true;
			
			// Perform Tagging
			String pageId = this.activity.getLocalClassName(); // required
			for (Product product : this.products) {
				String productId = product.getId(); // required
				String productName = product.getName(); // required
				String categoryId = product.getCategoryId(); // required
				String[] attributes = product.getAttributes(); // optional,null-able

				success = DigitalAnalytics.fireProductview(pageId, productId, productName,
						categoryId, attributes);
			}

			this.finish(success);

		} catch (Exception e) {
			Log.e(Constants.LOG_TAG, e.getMessage(), e);
			this.finish(false);
		}
	}

	private void finish(boolean success) {
		// This is for functional testing/user feedback only. In practice,
		// tagging should just be silent
	    Helper.myAlertDialog(activity, success, null, R.string.tag_product_view_success, R.string.tag_product_view_failed);
	}
}

File: SDKPackages/da/ios/Samples/KitchenSinkNew/KitchenSinkNew/TagProductView.m
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2016
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 ******************************************************************************/

#import "TagProductView.h"
#import "Product.h"
#import "DigitalAnalytics.h"

@interface TagProductView()

@property (nonatomic,strong) NSString *pageName;
@property (nonatomic,strong) NSArray *products;

@end

@implementation TagProductView

@synthesize pageName;
@synthesize products;

-(id) initTag:(NSString *) pageNameParam products:(NSArray *) productsParam
{
    self = [super init];
    
    if(self)
    {
        self.pageName = pageNameParam;
        self.products = productsParam;
    }
    
    return self;
}

-(void) executeTag
{
    BOOL success = YES;
    
    for(Product *product in products)
    {
        NSString *productId = product.productId;
        NSString *productName = product.name;
        NSString *categoryId = product.categoryId;
        NSArray *attributes = product.attributes;
        
        if(![DigitalAnalytics fireProductview:self.pageName productId:productId productName:productName category:categoryId attributes:attributes])
        {
            success = NO;
        }
    }
    
    [self finish:success];
}

@end
